helix-view/Cargo.toml

```diff
@@ -28,6 +28,7 @@ tokio = { version = "1", features = ["rt", "rt-multi-thread", "io-util", "io-std
 futures-util = { version = "0.3", features = ["std", "async-await"], default-features = false }
 
 slotmap = "1"
+rustc-hash = "1.1.0"
```

Comment:
Let's drop this extra dependency, the documents hashmap isn't worth optimizing because it's not a bottleneck.

Reply:
I added this because many places we use `HashMap` we can use `FxHashMap` instead and we can get free speed, for example in themes, registers, theme palettes, .etc which are accessed a lot. It is used in rustc and rust-analyzer so I thought it would make sense here. But maybe I should put that in a separate pr.

Reply:
While I agree that it's nice to use a fast hash function, there's still very few lookups compared to something like rustc which will do hundreds of thousands of lookups during compile time -- I don't think hashmaps even show up on the flamegraph for us.  (Note that `fx` is also optimized for integer keys)

Such an optimization needs benchmarking to back it up: speeding up an operation 80%~100% is great, but if that operation consumed less than 1% of total runtime, then the actual performance will be largely unchanged.